phoenix-core/src/main/java/org/apache/phoenix/compile/DeleteCompiler.java

```diff
@@ -872,6 +887,12 @@ public QueryPlan getQueryPlan() {
         }
     }
 
+    /**
+     * Implementation of MutationPlan that is selected if
+     * 1) there are immutable indexes presented for the table,
+     * 2) the query has WHERE clause and is not strictly point lookup, and
```

Comment:
Point 2 is applicable to ServerSelectPlan also.

Reply:
So I would recommend we move bullet 2 to serverselect as bullet 5. And just one bullet point here.